### PR TITLE
Minor: Move hash utils to common

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Closes #.

## Rationale for this change

While working on https://github.com/apache/arrow-datafusion/pull/7629, I found `create_hashes` is placed to physical-expr.

Hash utils only import functions from arrow, std, and df-common. It seems it should be placed to df-common not physical-expr.

## What changes are included in this PR?

We can have hash_utils from df-common.

## Are these changes tested?

Nope, only import is modified, no need to add additional tests.

### CODE REVIEW

File: datafusion/physical-expr/src/lib.rs [hunk 1]
```diff
@@ -28,7 +28,6 @@ pub mod equivalence;
 pub mod execution_props;
 pub mod expressions;
 pub mod functions;
-pub mod hash_utils;
```
Comment (alamb) [inline datafusion/physical-expr/src/lib.rs:31]: I think this is technically a breaking change if someone used `hash_utils` directly in their code.

Perhaps we could  leave a pointer like

```suggestion
// backwards compatibility
pub use datafusion_common::hash_util;
```

But I don't think that is critical

**Review by alamb — APPROVED** [review]
Thank you @jayzhan211 

I double checked and this change adds no new dependencies:
https://crates.io/crates/half/reverse_dependencies is already a dependency on arrow as is ahash: https://crates.io/crates/ahash/reverse_dependencies?page=3

**Comment by jayzhan211** [comment]
how to re-run CI without code changes?

**Comment by alamb** [comment]
> how to re-run CI without code changes?

You can go to the job and click rerun:

https://github.com/apache/arrow-datafusion/actions/runs/6345797930/job/17238371168?pr=7684

Screenshot 2023-09-29 at 9 52 45 AM

**Comment by jayzhan211** [comment]
Screenshot 2023-09-29 at 22 24 29
I dont have the option :(

**Comment by alamb** [comment]
> I dont have the option :(

It must be related to permissions. 

Another way to do so is to merge up from main (`git merge apache/main`) which I just did for this PR -- I'll plan to merge it once CI passes. Thank you

**Pull request merged**